pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -43,11 +45,27 @@ const (
 
 var runInTestMode bool
 var runInDemoMode bool
+var maxConcurrentReconciles int 
```

Comment:
nit: could we do `= 3` here so that nobody has to hunt through the code for the default?

Reply:
Funnily I had this many commits ago, but was asked to not specify any value during declaration. But I agree, its better to have a default here...